tests/components/evohome/test_water_heater.py

```diff
@@ -0,0 +1,214 @@
+
+    async for _ in setup_evohome(hass, config, install=install):
+        if (dhw := get_dhw_entity(hass)) is None:
+            pytest.skip("this installation has no DHW to test")
+
+        # turn_away_mode_off(): FollowSchedule
+        with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
+            await dhw.async_turn_away_mode_off()
+
+            assert mock_fcn.await_count == 1
+            assert mock_fcn.await_args.args == (
+                {
+                    "mode": "FollowSchedule",
+                    "state": None,
+                    "untilTime": None,
+                },
+            )
+            assert mock_fcn.await_args.kwargs == {}
+
+
+@pytest.mark.parametrize("install", TEST_INSTALLS)
+async def test_turn_away_mode_on(
+    hass: HomeAssistant,
+    config: dict[str, str],
+    install: str,
+) -> None:
+    """Test water_heater services of a evohome-compatible DHW zone."""
+
+    async for _ in setup_evohome(hass, config, install=install):
+        if (dhw := get_dhw_entity(hass)) is None:
+            pytest.skip("this installation has no DHW to test")
+
+        # turn_away_mode_on(): PermanentOverride, Off
+        with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
+            await dhw.async_turn_away_mode_on()
+
+            assert mock_fcn.await_count == 1
+            assert mock_fcn.await_args.args == (
+                {
+                    "mode": "PermanentOverride",
+                    "state": "Off",
+                    "untilTime": None,
+                },
+            )
+            assert mock_fcn.await_args.kwargs == {}
+
+
+@pytest.mark.parametrize("install", TEST_INSTALLS)
+async def test_turn_off(
+    hass: HomeAssistant,
+    config: dict[str, str],
+    install: str,
+) -> None:
+    """Test water_heater services of a evohome-compatible DHW zone."""
+
+    async for _ in setup_evohome(hass, config, install=install):
+        if (dhw := get_dhw_entity(hass)) is None:
+            pytest.skip("this installation has no DHW to test")
+
+        # turn_off(): PermanentOverride, Off
+        with patch("evohomeasync2.hotwater.HotWater._set_mode") as mock_fcn:
+            await dhw.async_turn_off()
+
+            assert mock_fcn.await_count == 1
+            assert mock_fcn.await_args.args == (
+                {
+                    "mode": "PermanentOverride",
+                    "state": "Off",
+                    "untilTime": None,
+                },
+            )
+            assert mock_fcn.await_args.kwargs == {}
+
+
+@pytest.mark.parametrize("install", TEST_INSTALLS)
+async def test_turn_on(
+    hass: HomeAssistant,
+    config: dict[str, str],
+    install: str,
+) -> None:
+    """Test water_heater services of a evohome-compatible DHW zone."""
+
+    async for _ in setup_evohome(hass, config, install=install):
+        if (dhw := get_dhw_entity(hass)) is None:
+            pytest.skip("this installation has no DHW to test")
```

Comment:
Is there a way we can omit using `if` in this test?

Reply:
Some fixtures have DHW, others not (as in the real world).

I could parameterize?
```
TEST_INSTALLS_WITH_DHW = [i for i in TEST_INSTALLS if has_dhw(i)]
   ...
@pytest.mark.parametrize("install", TEST_INSTALLS_WITH_DHW)
```

Reply:
... or:
```
TEST_INSTALLS_WITH_DHW: Final = (
    "default",  # evohome: multi-zone, with DHW
)
```

Reply:
Up to you 😁